File: 8 - Dictionaries and Maps/Solution.cs
using System;
using System.Collections.Generic;

class Solution
{
    static void Main(String[] args)
    {
        var n = int.Parse(Console.ReadLine());
        var phoneBook = new Dictionary<string, int>();

        for (var i = 0; i < n; i++)
        {
            var entry = Console.ReadLine().Split(' ');
            var name = entry[0];
            var phone = int.Parse(entry[1]);
            phoneBook.Add(name, phone);
        }

        for (var i = 0; i < n; i++)
        {
            var name = Console.ReadLine();
            if (phoneBook.ContainsKey(name))
            {
                var phone = phoneBook[name];
                Console.WriteLine($"{name}={phone}");
            }
            else Console.WriteLine("Not found");
        }
    }
}

File: 8 - Dictionaries and Maps/Solution.fs
open System
open System.Collections.Generic

[<EntryPoint>]
let main argv = 
    let n = Console.ReadLine() |> int
    let phoneBook = Dictionary<string, int>()
    for i = 0 to n - 1 do
        let entry = Console.ReadLine().Split(' ')
        let name = entry.[0]
        let phone = entry.[1] |> int
        phoneBook.Add(name, phone)
    for i = 0 to n - 1 do
        let name = Console.ReadLine()
        if phoneBook.ContainsKey(name) then 
            let phone = phoneBook.[name]
            printfn "%s=%i" name phone
        else printfn "Not found"
    0
